src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -304,17 +374,23 @@ public String getQuestionWithoutExistingResponseSubmissionFormHtml(
     @Override
     public String getQuestionSpecificEditFormHtml(int questionNumber) {
         StringBuilder optionListHtml = new StringBuilder();
-        String optionFragmentTemplate = FormTemplates.MSQ_EDIT_FORM_OPTIONFRAGMENT;
-        for (int i = 0; i < numOfMsqChoices; i++) {
-            String optionFragment =
-                    Templates.populateTemplate(optionFragmentTemplate,
-                            Slots.ITERATOR, Integer.toString(i),
-                            Slots.MSQ_CHOICE_VALUE, SanitizationHelper.sanitizeForHtml(msqChoices.get(i)),
-                            Slots.MSQ_PARAM_CHOICE, Const.ParamsNames.FEEDBACK_QUESTION_MSQCHOICE);
 
-            optionListHtml.append(optionFragment).append(Const.EOL);
+        if (generateOptionsFor == FeedbackParticipantType.NONE) {
+            String optionFragmentTemplate = FormTemplates.MSQ_EDIT_FORM_OPTIONFRAGMENT;
```

Comment:
Not entirely sure myself: why does this whole code block need to be moved to an `if` condition?

Reply:
When MSQ has options generated, then `numOfMsqChoices` is equal to number of entities from which it options are generated, i.e, number of students/instructors/teams. But we are not storing that information in `msqChoices` list. So, if that `if` clause weren't there, control will go inside [this](https://github.com/TEAMMATES/teammates/blob/master/src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java#L308) loop, and then we would get an exception for [this](https://github.com/TEAMMATES/teammates/blob/master/src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java#L312) line.

Reply:
Instead of the `if` statement, I am changing the loop on [this](https://github.com/TEAMMATES/teammates/blob/master/src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java#L308) line from -

``` java
for (int i = 0; i < numOfMsqChoices; i++) {
    // code
}
```

to

``` java
for (int i = 0; i < msqChoices.size(); i++) {
    // code
}
```